bundles/org.openhab.binding.carnet/src/main/java/org/openhab/binding/carnet/internal/handler/CarNetAccountHandler.java

```diff
@@ -0,0 +1,295 @@
+import org.openhab.binding.carnet.internal.api.CarNetApiGSonDTO.CarNetVehicleDetails;
+import org.openhab.binding.carnet.internal.api.CarNetApiGSonDTO.CarNetVehicleList;
+import org.openhab.binding.carnet.internal.api.CarNetHttpClient;
+import org.openhab.binding.carnet.internal.api.CarNetTokenManager;
+import org.openhab.binding.carnet.internal.config.CarNetAccountConfiguration;
+import org.openhab.binding.carnet.internal.config.CarNetCombinedConfig;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link CarNetAccountHandler} implements access to the myAudi account API. It is implemented as a brdige device
+ * and also dispatches events to the vehicle things.
+ *
+ * @author Markus Michels - Initial contribution
+ * @author Lorenzo Bernardi - Additional contribution
+ *
+ */
+@NonNullByDefault
+public class CarNetAccountHandler extends BaseBridgeHandler {
+    private final Logger logger = LoggerFactory.getLogger(CarNetAccountHandler.class);
+    private final CarNetCombinedConfig config = new CarNetCombinedConfig();
+    private final CarNetTokenManager tokenManager;
+    private final CarNetApi api;
+    private final CarNetHttpClient http;
+
+    private List<CarNetVehicleInformation> vehicleList = new ArrayList<>();
+    private List<CarNetDeviceListener> vehicleInformationListeners = Collections
+            .synchronizedList(new ArrayList<CarNetDeviceListener>());
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    /**
+     * Constructor
+     *
+     * @param bridge Bridge object representing a FRITZ!Box
+     */
+    public CarNetAccountHandler(Bridge bridge, @Nullable CarNetTextResources resources,
+            CarNetTokenManager tokenManager) {
+        super(bridge);
+        this.tokenManager = tokenManager;
+
+        // Each instance has it's own http client. Audi requires weaked SSL attributes, other may not
+        HttpClient httpClient = new HttpClient();
+        try {
+            SslContextFactory ssl = new SslContextFactory();
+            // ssl.setIncludeCipherSuites("^TLS_RSA_.*$");
+            String[] excludedCiphersWithoutTlsRsaExclusion = Arrays.stream(ssl.getExcludeCipherSuites())
+                    .filter(cipher -> !cipher.equals("^TLS_RSA_.*$")).toArray(String[]::new);
+            ssl.setExcludeCipherSuites(excludedCiphersWithoutTlsRsaExclusion);
+            httpClient = new HttpClient(ssl);
+            httpClient.start();
+            logger.debug("{}", httpClient.dump());
+        } catch (Exception e) {
+            logger.warn("Unable to start HttpClient!");
```

Comment:
You might want to log the exception's message.

Reply:
done